gap/congruences/conglatt.gi

```diff
@@ -125,19 +115,19 @@ SEMIGROUPS.PrincipalXCongruencePosetNC :=
   SEMIGROUPS.OptionsRec(S).report := report;
 
   # We are done: make the object and return
-  po := Digraph(children);
-  SetInNeighbours(po, parents);
-  poset := Objectify(NewType(FamilyObj(children), IsCongruencePoset),
-                     rec(po := po));
+  poset := Digraph(children);
+  SetInNeighbours(poset, parents);
   SetCongruencesOfPoset(poset, congs);
+  SetDigraphVertexLabels(poset, congs);
   SetUnderlyingSemigroupOfCongruencePoset(poset, S);
+  SetFilterObj(poset, IsCongruencePoset);
   return poset;
 end;
 
 InstallMethod(MinimalCongruences,
 "for a congruence poset",
 [IsCongruencePoset],
-poset -> CongruencesOfPoset(poset){DigraphSinks(poset!.po)});
+poset -> CongruencesOfPoset(poset){DigraphSinks(poset)});
```

Comment:
When you add loops at every vertex, you'll have to change this. Options: vertices with out-degree 1, or vertices with in-degree 1. Or remove loops and then get the sinks.